### PR TITLE
* add autochecking logic to CLI to detect out of sync configuration i…

### PR DESCRIPTION
**Pull request by mayitbeegh**
…n update-stack or view-config

### CODE REVIEW

File: src/main/java/com/nike/cerberus/store/ConfigStore.java [hunk 1]
```diff
@@ -739,4 +739,52 @@ public void setAuditLoggingEnabled(boolean auditLoggingEnabled) {
...
+     */
+    public boolean isConfigSynchronized(){
+        EnvironmentData environmentData = getDecryptedEnvironmentData();
+        Map<Regions, RegionData> map = environmentData.getRegionData();
+        String lastProperties = null;
+        String lastEnvironmentData = null;
+        Regions lastRegions = null;
+        boolean result = true;
+        for (Map.Entry<Regions, RegionData> entry: map.entrySet()){
+            Regions currentRegions = entry.getKey();
```
Comment (sdford) [inline src/main/java/com/nike/cerberus/store/ConfigStore.java]: nitpick: this is really one "currentRegion", even though the class type is named plural as "Regions"

File: src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java [hunk 2]
```diff
@@ -126,6 +126,7 @@ public void run(UpdateStackCommand command) {
 
     @Override
     public boolean isRunnable(UpdateStackCommand command) {
+        configStore.isConfigSynchronized();
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java]: If there is a region outage, can I still run a command?

Comment (fieldju) [reply src/main/java/com/nike/cerberus/operation/core/UpdateStackOperation.java]: Also why are you running this in multiple places, could you not do this in a single global place for all commands?

File: src/main/java/com/nike/cerberus/store/ConfigStore.java [hunk 3]
```diff
@@ -739,4 +739,59 @@ public void setAuditLoggingEnabled(boolean auditLoggingEnabled) {
     public boolean isAuditLoggingEnabled() {
         return getEnvironmentData().isAuditLoggingEnabled();
     }
+
+    /***
+     * Returns whether configs (environment.json, cms/environment.properties) are the same in the regions defined in
+     * environment.json
+     * @return False if the configs are different and true if the configs are the same
+     */
+    public boolean isConfigSynchronized(){
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/store/ConfigStore.java]: I would like to see unit tests for this code.

File: src/main/java/com/nike/cerberus/store/ConfigStore.java [hunk 4]
```diff
@@ -739,4 +739,59 @@ public void setAuditLoggingEnabled(boolean auditLoggingEnabled) {
...
+        EnvironmentData environmentData = getDecryptedEnvironmentData();
+        Map<Regions, RegionData> map = environmentData.getRegionData();
+        Properties lastProperties = null;
+        String lastEnvironmentData = null;
+        Regions lastRegions = null;
+        boolean result = true;
+        for (Map.Entry<Regions, RegionData> entry: map.entrySet()){
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/store/ConfigStore.java]: Does this work if there are 3 regions?

**Pull request merged**